src/components/can_cooperation/src/can_module.cc

```diff
@@ -336,5 +349,20 @@ void CANModule::RemoveAppExtension(uint32_t app_id) {
   }
 }
 
+bool CANModule::IsAppForPlugin(
+application_manager::ApplicationSharedPtr app) {
+  if (app->app_types()) {
+    std::vector<int> hmi_types =
+    application_manager::Service::SmartObjToArrayInt(app->app_types());
```

Comment:
This will add dependency on smartObjects to functional module. I'd rather avoid it.